pkg/clustermanager/cluster_manager.go

```diff
@@ -221,6 +216,24 @@ func (c *ClusterManager) CreateWorkloadCluster(ctx context.Context, managementCl
 		return nil, fmt.Errorf("error generating workload kubeconfig: %v", err)
 	}
 
+	if features.IsActive(features.TaintsSupport()) {
```

Comment:
I would suggest moving this flag somewhere else (maybe in validations?), and return an error if taints are specified while the feature is off
That way, we don't even accept configs that have taints